rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -168,6 +175,11 @@ void SqliteStorage::fill_topics_and_types()
   }
 }
 
+std::string SqliteStorage::get_database_name(const std::string & uri)
+{
+  return rosbag2_storage::FilesystemHelper::get_folder_name(uri) + ".db3";
```

Comment:
What happens if I put a `.db3` file in a new folder and want to open this one? LIke shifting backfiles around?
```
.../my_new_folder/my_old_bag.db3
```

Reply:
At the moment it is required that the database file is named like the bag folder.

We can change this behavior as soon as the metadata file is available (PR #45).
After that we should use the file path provided by the metadata file to locate the database file.
This path then has to be adjusted after renaming/moving the database file.

Reply:
> Like shifting bagfiles around?
Generally, we should consider the folder being the bagfile. 
So the user should ideally never move/rename/modify the files inside a bag, except when he knows exactly what he does.

Reply:
I somewhat agree with it, but I also don't see a strong reason why we should enforce it. I could imagine having an old bagfile around and put it into a new folder. Then I should still be able to load this file.
Also in the long term run, whenever there are split rosbag files, this strong naming match won't hold anymore as every individual bag file will have some sort of pre-/postfix attached to it.

Just for now, I don't see why we should enforce this on code level. Given a URI, I should always be able to extract the folder name independant from the database file.

Reply:
> I somewhat agree with it, but I also don't see a strong reason why we should enforce it. I could imagine having an old bagfile around and put it into a new folder. Then I should still be able to load this file.

The problem is that we do not have any metadata at this point.
So I see no way how rosbag2 should be able to know that a file in a folder can be understood as a database file.
That's  the reason why we enforce this naming scheme for now.
As soon as the info PR lands we have the metadata file which contains the relative file path to the database and can use this path to locate the database.

> Also in the long term run, whenever there are split rosbag files, this strong naming match won't hold anymore as every individual bag file will have some sort of pre-/postfix attached to it.

This is also solved using the relative file paths from the metadata file. 
This case is the reason why we added the `relative_file_paths` array to the metadata file.

> Just for now, I don't see why we should enforce this on code level. Given a URI, I should always be able to extract the folder name independant from the database file.

Since the database file naming is solely handled by the storage plugin I don't see how this can be achieved through the general rosbag2 interface.